flang/lib/Semantics/semantics.cpp

```diff
@@ -655,8 +655,11 @@ bool Semantics::Perform() {
 void Semantics::EmitMessages(llvm::raw_ostream &os) {
   // Resolve the CharBlock locations of the Messages to ProvenanceRanges
   // so messages from parsing and semantics are intermixed in source order.
+  bool echoSourceLine{true};
+  const auto &features{context_.languageFeatures()};
   context_.messages().ResolveProvenances(context_.allCookedSources());
-  context_.messages().Emit(os, context_.allCookedSources());
+  context_.messages().Emit(
+      os, context_.allCookedSources(), echoSourceLine, &features);
```

Comment:
`echoSourceLine` doesn't really need to be a variable, does it?  If not, `/*echoSourceLine=*/true` might be more straightforward.